app/client-v2/packages/ui/components/MapByLayer.tsx

```diff
@@ -290,24 +312,30 @@ export default function MapByLayer(props: MapProps) {
         <Layer type="raster" id={stylesObj["xyz"].id} source={stylesObj["xyz"].id} />
       </Source>
     ),
+    wms: (
+      <Source id={stylesObj["xyz"].id} type="raster" tiles={[stylesObj["wms"].url]} tileSize={256}>
+        <Layer type="raster" id={stylesObj["xyz"].id} source={stylesObj["xyz"].id} />
+      </Source>
+    )
   };
 
   return (
     <div style={{ width: "100%", height: "100vh" }}>
-      <Map
-        initialViewState={{
-          latitude: stylesObj[layer].center[1],
-          longitude: stylesObj[layer].center[0],
-          zoom: stylesObj[layer].zoom,
-        }}
-        style={{ width: "100%", height: "100%" }}
-        mapStyle="mapbox://styles/mapbox/streets-v11"
-        mapboxAccessToken={MAPBOX_TOKEN}>
-        {layer === "poi" && sourcesObj.poi}
-        {layer === "aoi" && sourcesObj.aoi}
-        {layer === "edge" && sourcesObj.edge}
-        {layer === "xyz" && sourcesObj.xyz}
-      </Map>
+        <Map
+          initialViewState={{
+            latitude: stylesObj[layer].center[1],
+            longitude: stylesObj[layer].center[0],
+            zoom: stylesObj[layer].zoom,
+          }}
+          style={{width: "100%", height: "100%"}}
+          mapStyle="mapbox://styles/mapbox/streets-v11"
+          mapboxAccessToken={MAPBOX_TOKEN}>
+          {layer === "poi" && sourcesObj.poi}
+          {layer === "aoi" && sourcesObj.aoi}
+          {layer === "edge" && sourcesObj.edge}
+          {layer === "xyz" && sourcesObj.xyz}
+          {layer === "wms" && sourcesObj.wms}
+        </Map>
```

Comment:
I think it would be better to have this part based on the type of layer. "poi" is "point", "edge" is "line" and "aoi" is "fill". These are the three types for features layers. 
WMS and 'XYZ" are different layers so there we will not have types.

Reply:
fixed